TLCGen.PluggedInItems/TLCGen.Generators.CCOL/CodeGeneration/Functionality/PtpCodeGenerator.cs

```diff
@@ -206,20 +237,48 @@ public override string GetCode(ControllerModel c, CCOLCodeTypeEnum type, string
             switch (type)
             {
                 case CCOLCodeTypeEnum.RegCIncludes:
-                    if (c.PTPData.PTPKoppelingen.Count > 0)
+                    if ((c.PTPData.PTPKoppelingen.Count > 0) && c.PTPData.PTPKoppelingen.Any(x => x.Dummy != true))
                     {
                         sb.AppendLine($"{ts}#include \"{c.Data.Naam}ptp.c\" /* PTP seriele koppeling */");
                     }
                     return sb.ToString();
+                case CCOLCodeTypeEnum.RegCPreApplication:
+                    if ((c.PTPData.PTPKoppelingen.Count > 0) && c.PTPData.PTPKoppelingen.Any(x => (x.Dummy == true) && (x.MaakIO == true)))
+                    {
+                        sb.AppendLine($"{ts}/* overbrengen ingangen naar hulpelementen tbv koppeling */");
+                    }
+                    foreach (var k in c.PTPData.PTPKoppelingen.Where(x => (x.Dummy == true) && (x.MaakIO == true)))
+                    {
+                        for (var i = 1; i <= k.AantalsignalenIn; ++i)
+                        {
+                            sb.Append($"{ts}IH[{_hpf}{k.TeKoppelenKruispunt}{_hiks}" + i.ToString("00") + "]");
+                            sb.AppendLine($" = IS[{_ispf}{k.TeKoppelenKruispunt}{_isiks}" + i.ToString("00") + "];");
+                        }
+                        sb.AppendLine();
+                    }
+                    return sb.ToString();
+                case CCOLCodeTypeEnum.RegCPostApplication:
+                    if ((c.PTPData.PTPKoppelingen.Count > 0) && c.PTPData.PTPKoppelingen.Any(x => (x.Dummy == true) && (x.MaakIO == true)))
```

Comment:
de check op Count > 0 kan hier weg, die zit impliciet reeds in het Any() statement